### PR TITLE
chore(sdk): add topic sdk test

### LINKED ISSUE #2327: Adding topic sdk tests
### Feature Spec

The test will be written in Wing. 
it will cover the following methods of the Topic resource:
- [ ] publish
- [ ] on_message

Please be minded to test the methods with all possible arguments compositions, and to edge cases ✨ 

The tests should be placed under `examples/tests/sdk_tests/topic` 

### Use Cases

testing

### Implementation Notes

_No response_

### Component

SDK

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by marciocadev**
sdk test for `cloud.Topic`

Closes #2327

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

File: examples/tests/sdk_tests/topic/on_message.w [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+t.on_message(inflight() => {
+  c.inc();
+});
+
+t.on_message(inflight() => {
+  c.inc();
+});
+
+new cloud.Function(inflight() => {
```
Comment (revitalbarletz) [inline examples/tests/sdk_tests/topic/on_message.w:14]: @marciocadev I think that a race condition might occur here - take a look at the queue tests implementation and see the trick with the Predicate class:

inflight assert(predicate: Predicate) { 
    let var i = 0;
    while i<600 {
      i = i + 1;
      if predicate.test() {
        assert(predicate.test());
        return;
      } 
      TestHelper.sleep(100);
    }
    assert(predicate.test());
  }

File: examples/tests/sdk_tests/topic/publish.w [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+let t = new cloud.Topic();
+let c = new cloud.Counter();
+
+t.on_message(inflight(e: str) => {
+  assert(e == "1");
+  c.inc();
+});
+
+new cloud.Function(inflight() => {
+  t.publish("1");
```
Comment (revitalbarletz) [inline examples/tests/sdk_tests/topic/publish.w:12]: Same comment here, a race condition might occur.

**Review by tsuf239 — APPROVED** [review]
Amazing work!! thanks :)
